spec/room_spec.rb

```diff
@@ -0,0 +1,167 @@
+require "simplecov"
+SimpleCov.start
+
+require_relative "spec_helper"
+
+describe "Room class" do
+  before do
+    @room = Hotel::Room.new(room_number: 1)
+  end
+  it "creates an instance of Room" do
+    expect(@room).must_be_kind_of Hotel::Room
+  end
+
+  it "includes an array when initialized" do
+    expect(@room.reservations).must_be_kind_of Array
+  end
+
+  it "includes a cost per night" do
+    expect(@room.cost_per_night).must_equal 200.00
+  end
+
+  describe "add_reservation method" do
+    before do
+      @room = Hotel::Room.new(room_number: 1)
+      @reservation = Hotel::Reservation.new(check_in: "2019-3-30", check_out: "2019-4-1", room: @room)
+    end
+
+    it "adds the reservation" do
+      expect(@room.reservations).wont_include @reservation
+      previous = @room.reservations.length
+
+      @room.add_reservation(@reservation)
+
+      expect(@room.reservations).must_include @reservation
+      expect(@room.reservations.length).must_equal previous + 1
+    end
+  end
+  describe "is_available? method" do
+    before do
+      @manager = Hotel::Manager.new
+      @check_in = "2020-3-20"
+      @check_out = "2020-3-25"
+
+      @reservation = @manager.reserve_room(@check_in, @check_out)
+      @reserved_room = @reservation.room
+    end
+
+    it "returns false if the given date range STARTS and ENDS the same day as a reservation" do
+      check_in = Date.parse(@check_in)
+      check_out = Date.parse(@check_out)
+      date_range = (check_in..check_out)
+      availability = @reserved_room.is_available?(date_range)
+
+      expect(availability).must_equal false
+    end
+
+    it "returns false if the given date range STARTS on reservaton's check_in day and ENDS DURING" do
+      check_in = Date.parse(@check_in)
+      check_out = Date.parse("2020-03-22")
+      date_range = (check_in..check_out)
```

Comment:
Good work laying out all of these different test cases.

Since you recreate them so frequently, it might make sense to store the instances of `Date` as instance variables in the `before` block as well. Then in the tests, you could do something like:

```ruby
date_range = @check_in_date..(@check_out_date - 2)
```